### PR TITLE
[wasm] Handle VoidTaskResult for async Task

### LINKED ISSUE #47780: [wasm] plain Task marshaling failing
The bindings are failing when mashing a non generic task

Reproduce with the wasm console sample and a

```

using System;
using System.Threading.Tasks;

public class Test
{
    public static async Task Main(string[] args)
    {
        await Task.Delay(1);
        Console.WriteLine("Hello World!");
        for (int i = 0; i < args.Length; i++) {
            Console.WriteLine($"args[{i}] = {args[i]}");
        }
        return;
    }
}
```

Error is
```
Error: no idea on how to unbox value types
    at Object._unbox_mono_obj_rooted_with_known_nonprimitive_type (dotnet.js:1:175395)
    at Object._unbox_mono_obj_rooted (dotnet.js:1:176729)
    at Object._mono_array_to_js_array_rooted (dotnet.js:1:173432)
    at Object.mono_array_to_js_array (dotnet.js:1:172973)
    at Object.mono_wasm_parse_args (dotnet.js:1:205340)
    at _mono_wasm_invoke_js_with_args (dotnet.js:1:211789)
    at <anonymous>:wasm-function[10855]:0x1a848d
    at <anonymous>:wasm-function[3406]:0x7df7e
    at <anonymous>:wasm-function[2240]:0x488b7
    at <anonymous>:wasm-function[8112]:0x1423cf
```

### PR DESCRIPTION
**Pull request by lewing**
fixes https://github.com/dotnet/runtime/issues/47780

### CODE REVIEW

**Review by kg — APPROVED** [review]

**Review by kjpou1 — APPROVED** [review]

File: src/mono/wasm/runtime/driver.c [hunk 1]
```diff
@@ -823,6 +826,10 @@ void mono_wasm_ensure_classes_resolved ()
 		safehandle_class = mono_class_from_name (mono_get_corlib(), "System.Runtime.InteropServices", "SafeHandle");
 		resolved_safehandle_class = 1;
 	}
+	if (!voidtaskresult_class && !resolved_voidtaskresult_class) {
+		voidtaskresult_class = mono_class_from_name (mono_get_corlib(), "System.Threading.Tasks", "VoidTaskResult");
```
Comment (stephentoub) [inline src/mono/wasm/runtime/driver.c:830]: This is fragile.  There's no way to support any `Task<T>` regardless of its type parameter?

Comment (kg) [reply src/mono/wasm/runtime/driver.c:830]: VoidTaskResult is a struct, we don't handle structs. (I'm working on that)

Comment (stephentoub) [reply src/mono/wasm/runtime/driver.c:830]: > I'm working on that

Ok

Comment (lewing) [reply src/mono/wasm/runtime/driver.c:830]: Yeah, the less fragile fix in in process. I did the simple fix after digging into it so we can unblock/write some tests.

**Pull request merged**